app/core/middleware/logger.py

```diff
@@ -1,59 +1,125 @@
-    # log space for no benefit.
+    def __init__(self, app: FastAPI, *, logger: logging.Logger, file_logger: logging.Logger = None) -> None:
+        self._logger = logger
+        self._file_logger = file_logger
+        super().__init__(app)
 
     # https://github.com/tiangolo/fastapi/issues/394#issuecomment-927272627
-    # async def set_body(self, request: Request):
-    #     receive_ = await request._receive()
+    async def set_body(self, request: Request):
+        receive_ = await request._receive()
 
-    #     async def receive() -> Message:
-    #         return receive_
+        async def receive() -> Message:
+            return receive_
 
-    #     request._receive = receive
+        request._receive = receive
 
     async def dispatch(self, request, call_next):
-            start_time = time.time()
+        if request.url.path.startswith("/api/v1/health"):
+            response = await self._execute_request(call_next, request)
+            return
+        
+        request_id: str = str(uuid4())
+        logging_dict = {
+            "X-API-REQUEST-ID": request_id  # X-API-REQUEST-ID maps each request-response to a unique ID
+        }
 
-            # await self.set_body(request)
-            # req_body = await request.body()
-            response = await call_next(request)
+        await self.set_body(request)
+        response, response_dict = await self._create_response_log(
+            call_next,
+            request,
+            request_id
+        )
+        request_dict = await self._create_request_log(request)
+        logging_dict["req"] = request_dict
+        logging_dict["res"] = response_dict
 
-            # Consuming FastAPI response and grabbing body here
-            # resp_body = [section async for section in response.__dict__['body_iterator']]
-            # Repairing FastAPI response
-            # response.__setattr__('body_iterator', aiwrap(resp_body))
+        self._logger.info(logging_dict)
+        if self._file_logger is not None:
+            resp_body = [section async for section in response.__dict__["body_iterator"]]
+            response.__setattr__("body_iterator", AsyncIteratorWrapper(resp_body))
 
-            # Formatting response body for logging
-            # try:
-            #     resp_body = json.loads(resp_body[0].decode())
-            # except:
-            #     resp_body = str(resp_body)
+            try:
+                resp_body = json.loads(resp_body[0])
+            except:
+                resp_body = str(resp_body)
 
-            request.app.logger.info(
+            response_dict["body"] = resp_body
+            logging_dict["res"] = response_dict
+
+            self._file_logger.debug(logging_dict)
+
+        return response
+    
+    async def _create_request_log(self, request: Request) -> str:
+
+        path = request.url.path
+        if request.query_params:
+            path += f"?{request.query_params}"
```

Comment:
why does this exist?

Reply:
So that query params are included in the log. I forgot to change line 73 to use `path` instead of `request.url.path`, so I can change that. Do you think it's not necessary?

Reply:
Path isn't used anywhere or am I missing something?

Reply:
No it's just not being used, yes.

Reply:
I'm not sure why it's not showing up in GH but it's used on line 73 instead of `request.url.path`